Delete boehm specific code

jacob-hughes opened the pull request:
This is now being served from the softdev/boehm_allocator crate, which
contains code needed for both rboehm and rustc_boehm.

jacob-hughes committed:
Delete boehm specific code

This is now being served from the softdev/libgc_internal crate, which
contains code needed for both libgc and rustgc.

ltratt commented:
A maybe-silly question: why does this repo exist? Or, maybe accurately, is its name misleading or ...? I'm a bit confused that we now have rustc_boehm, rboehm, and boehm_allocator repositories and no longer entirely sure which does which :)

jacob-hughes committed:
Change project name from rboehm => libgc

The Boehm GC is just one of many potential collector implementations, so
lets try and move away from referring to it in part of the name.

jacob-hughes committed:
Expand README to explain repo structure

jacob-hughes commented:
I've pushed a README that explains the structure of the GC library a bit better. I've also renamed uses of rboehm => libgc in source.

ltratt commented on README.md:
```diff
@@ -1,2 +1,31 @@
+      implementation: the Boehm-Demers-Weiser GC. Users should never interact
+      directly with this crate. Instead, any relevant APIs are re-exported
+      through libgc.
+    * rustgc: a fork of rustc with GC-aware optimisations. This can be used to
```
Should we say something like "Use of rustgc is not mandated, but it enables further optimisations for programs which use `libgc`"?

jacob-hughes replied:
Good idea, added here: 19fc1c185090d732dc5ab852fcdc689426bb194e (Also fixed the markdown syntax)

ltratt commented:
Please squash.

jacob-hughes commented:
Squashed, but we can't merge until https://github.com/softdevteam/libgc_internal/pull/1 is merged.

ltratt commented:
bors r+

jacob-hughes commented:
Format fixes: dd73b30a973b434b4055c2386787d37040eb309f

ltratt commented:
Please squash.

jacob-hughes commented:
Fixed

ltratt commented:
bors r+
